### PR TITLE
Sites: redirect to onboarding flow when browsing the page with no sites

### PR DESCRIPTION
**Pull request by zaguiini**
Related to pet6gk-xy-p2#comment-475.

## Proposed Changes

We can remove a point of friction from the onboarding flow if we simply redirect the user to `/setup/new-hosted-site` if they're browsing `/sites` with zero sites in their account. I added a link to `/setup/import-hosted-site` so that they can still import their sites if they want to.

image

If they are importing a site, they will also get the link to create a new one instead:

image

"Why not redirect to the URL directly after signing up?"

This is a good question. I believe sending them to sites and then deciding what their flow should be would allow us to centralize the decision to a single place. But that's only me, I have no hard feelings about modifying the destination from `/start/hosting`. I was thinking about potential bounces too: if the user ends up browsing `/sites` in the future without going through the login step, they can start from where they left off providing they signed up as a developer.

## Testing Instructions

In an account with no sites, browse `/sites?hosting-flow=true` and you should be redirected to `/setup/new-hosted-site?hosting-flow=true`. Verify that "Import a site" link redirects you to `/setup/import-hosted-site?hosting-flow=true`.

You can browse `/sites` without the query parameter and verify that you'll see the regular onboarding flow if you're not a developer.

### CODE REVIEW

**Comment by zaguiini** [comment]
I'm seeing a weird behavior where `page.redirect` and `page.replace` are changing the URL but the page is blank 🤔 Not sure what's going on here.

File: client/sites-dashboard/controller.tsx [hunk 1]
```diff
@@ -47,53 +47,21 @@ export function sanitizeQueryParameters( context: PageJSContext, next: () => voi
...
 
 	if ( context.query[ 'hosting-flow' ] || siteCount === 0 ) {
-		context.store.dispatch( hideMasterbar() );
-		return emptySites( context, siteCount, next );
+		const newSiteUrl = getAddNewSiteUrl( {
+			isHostingFlow: context.query[ 'hosting-flow' ] === 'true',
+			isDevAccount: getUserSetting( state, 'is_dev_account' ) === true,
+		} );
+
+		return window.location.assign( newSiteUrl );
```
Comment (zaguiini) [inline client/sites-dashboard/controller.tsx:59]: This creates a full-page refresh. Don't like it. Using `page.redirect` or `page.replace` doesn't work for some reason: I get a blank page with no errors in the console. 🤔

**Comment by vishnugopal** [comment]
How will folks import when they don't have a site address & only an export file in this flow?

Am personally ok with this change otherwise, seems to make more sense for the user & we always have an in-product import flow.

**Comment by p-jackson** [comment]
Some comments about why I'm not a fan pet6gk-xy-p2#comment-477

**Comment by zaguiini** [comment]
Not moving forward with this approach: pet6gk-xy-p2#comment-480

**Pull request closed without merging**